### PR TITLE
feat: show correct denom on sell order table and buy order modal

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1157 regen-network/regen-registry#1159
This PR depends on regen-js PR: https://github.com/regen-network/regen-js/pull/40

**Note**: this PR has not been tested yet since we don't have IBC coins on redwood yet, but the code is ready to be reviewed.

Add:
- new coins assets (axlUSDC, gravUSDC, EEUR) that will likely be added by governance on mainnet
- new component `DenomIcon` to display a coin logo based on a denom
- new component `DenomLabel` to display a coin label based on a denom
- these components in the sell order table and buy order modal
- a prefix ($/€) based on the denom used
- IBC coins support in `useQuerySellOrders` by unwrapping IBC hash with `DenomTrace`
- new API file in lib under `lib/transfer/api` with methods for `DenomTrace` and `DenomTraces`
---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/components/molecules/DenomIcon/DenomIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import GravUsdcIcon from 'web-components/lib/components/icons/coins/GravUsdcIcon';
+import { RegenTokenIcon } from 'web-components/lib/components/icons/RegenTokenIcon';
+
+export interface Props {
+  denom: string;
+  sx?: SxProps<Theme>;
+  iconSx?: SxProps<Theme>;
+}
+
+const DenomIcon = ({ denom, sx = [], iconSx = [] }: Props): JSX.Element => {
```
Comment (flagrede) [inline web-registry/src/components/molecules/DenomIcon/DenomIcon.tsx:20]: This component can be reused wherever we need to display a coin logo based on denom.

File: web-registry/src/components/molecules/DenomLabel/DenomLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { Label, LabelProps } from 'web-components/lib/components/typography';
+
+import { DENOM_LABEL_MAPPING } from './DenomLabel.config';
+
+export interface Props extends LabelProps {
+  denom: string;
+  sx?: SxProps<Theme>;
+}
+
+const DenomLabel = ({ denom, size, sx = [] }: Props): JSX.Element => {
```
Comment (flagrede) [inline web-registry/src/components/molecules/DenomLabel/DenomLabel.tsx:12]: Same as `DenomIcon` but for label.

**Review by haveanicedavid — APPROVED** [review]
lgtm

**Comment by erikalogie** [comment]
Great work on this!

A few things:
- It seems that the currency denom is displayed differently in different places: https://monosnap.com/file/qtq1STQhp8EHZersYcR9zk024fKLgY
- Some issue with the denom icon display. Should I maybe compress these in Illustrator and send you some new icons? https://monosnap.com/file/jetwbCQpFJWP8DOTp4j6GL5CAR2NYu
- Sometimes the "$" is missing: https://monosnap.com/file/LAp7o8TpsHGKbm0xVdrhvb8FrTx9pE

**Comment by flagrede** [comment]
@erikalogie all issues have been fixed.
- the denom label now always uses the governance-approved label
- $ should now always appear
- I no longer have the glitch with axelar usdc icon, let me know if you still see it.

**Review by victordelval — APPROVED** [review]
Wow, this is amazing! It looks so good. I haven't tested it locally yet, but I trust you a lot here ;) Thank you very much

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**